sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Linq/DigitalTwinsQueryBuilder.cs

```diff
@@ -0,0 +1,406 @@
+        public DigitalTwinsQueryBuilder<T> WhereCustom(string filter)
+        {
+            if (!string.IsNullOrEmpty(filter))
+            {
+                _clauses ??= new List<string>();
+                _clauses.Add(filter);
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a WHERE clause to a query along with conditions defined with formatted strings.
+        /// </summary>
+        /// <param name="filter">Conditional argument relevant to a DigitalTwins query (e.g. comparisons, DigitalTwins Functions, etc.)</param>
+        /// <returns>Query that contains a WHERE clause and conditional arguments.</returns>
+        public DigitalTwinsQueryBuilder<T> Where(FormattableString filter) => Where(filter, null);
+
+        /// <summary>
+        /// Adds a WHERE clause to a query along with conditions defined with LINQ expressions.
+        /// </summary>
+        /// <param name="filter">Conditional argument relevant to a DigitalTwins query (e.g. comparisons, DigitalTwins Functions, etc.)</param>
+        /// <param name="formatProvider">Culture-specific formatting information for DigitalTwins queries.</param>
+        /// <returns>Query that contains a WHERE clause and conditional arguments.</returns>
+        public DigitalTwinsQueryBuilder<T> Where(FormattableString filter, IFormatProvider formatProvider)
+        {
+            formatProvider ??= CultureInfo.InvariantCulture;
+
+            if (filter == null)
+            {
+                throw new ArgumentNullException(nameof(filter));
+            }
+
+            string[] args = new string[filter.ArgumentCount];
+
+            for (int i = 0; i < args.Length; i++)
+            {
+                args[i] = DigitalTwinsFunctions.Convert(filter.GetArgument(i), formatProvider);
+
+                if (args[i] == "null")
+                {
+                    throw new InvalidOperationException();
+                }
+            }
+
+            string text = string.Format(formatProvider, filter.Format, args);
+            return Where(text);
+        }
+
+        /// <summary>
+        /// Adds a WHERE clause to a query along with conditions defined with LINQ expressions.
+        /// </summary>
+        /// <param name="filter">Conditional argument relevant to a DigitalTwins query (e.g. comparisons, DigitalTwins Functions, etc.)</param>
+        /// <returns>Query that contains a WHERE clause and conditional arguments.</returns>
+        public DigitalTwinsQueryBuilder<T> Where(Expression<Func<T, bool>> filter)
+        {
+            string query = string.Empty;
+            Expression e = Evaluator.PartialEval(filter, CanEvaluate);
+            e = ExpressionNormalizer.Normalize(e, new Dictionary<Expression, Expression>());
+
+            LambdaExpression l = e as LambdaExpression;
+            Ensure(l != null);
+            Ensure(l.Parameters.Count == 1);
```

Comment:
Nit - consider adding messages to these which will make debugging easier:
```suggestion
            Ensure(l.Parameters.Count == 1, $"{nameof(Where)} expects a lambda expression with a single parameter.");
```